src/Avalonia.Controls/Application.cs

```diff
@@ -364,20 +369,28 @@ protected override void OnPropertyChanged(AvaloniaPropertyChangedEventArgs chang
 
             if (change.Property == RequestedThemeVariantProperty)
             {
-                if (change.GetNewValue<ThemeVariant>() is {} themeVariant && themeVariant != ThemeVariant.Default)
-                    SetValue(ActualThemeVariantProperty, themeVariant);
-                else
-                    ClearValue(ActualThemeVariantProperty);
+                if (change.GetNewValue<ThemeVariant>() is { } themeVariant)
+                {
+                    if (themeVariant == ThemeVariant.Default && PlatformSettings?.GetColorValues() is { } currentColors)
+                    {
+                        SetValue(ActualThemeVariantProperty, (ThemeVariant)currentColors.ThemeVariant);
+                    }
+                    else
+                    {
+                        SetValue(ActualThemeVariantProperty, themeVariant);   
+                    }
+                }
             }
             else if (change.Property == ActualThemeVariantProperty)
             {
                 ActualThemeVariantChanged?.Invoke(this, EventArgs.Empty);
             }
         }
-        
+
         private void OnColorValuesChanged(object? sender, PlatformColorValues e)
         {
-            SetValue(ActualThemeVariantProperty, (ThemeVariant)e.ThemeVariant, BindingPriority.Template);
+            if ((RequestedThemeVariant ?? ThemeVariant.Default) == ThemeVariant.Default)
+                SetCurrentValue(ActualThemeVariantProperty, (ThemeVariant)e.ThemeVariant);
```

Comment:
With this change, we break theme switching (at least on desktop), when:
1. Apps opens with default Dark variant.
1. User switches to non-default Dark variant.
2. OS changes global theme variant to Light variant. - at this point Application won't receive any changes about system theme because of this condition.
3. User switches back to Default variant. But app won't be changed, because previous OnColorValuesChanged was ignored.

We probably need to save latest value in the field. And use it when user changes to Default.

Reply:
@emmauss I pushed a commit that changes 372 line. When user switches to the Default variant - we now read GetColorValues(). Works on Windows well.

Let me know if it still works on android as expected.

Reply:
The changes on 372 prevent Default from ever being sent to the toplevel impl, which we need on the android side.

Reply:
> @emmauss I pushed a commit that changes 372 line. When user switches to the Default variant - we now read GetColorValues(). Works on Windows well.
> 
> Let me know if it still works on android as expected.

It doesn't really work. Since only Light or Dark is ever sent to the OS with your change, the OS never switches to FollowSystem.
We could pass both Actual variant and Requested variant. Requested for the OS to know what the app want, and Actual for the OS to update the system bars and nav bar